Makefile

```diff
@@ -10,20 +10,39 @@ endif
 .PHONY: all
 all: codegen code
 
+.PHONY: ipv6code generics codegen codegen-%
+codegen: ipv6code generics $(addprefix codegen-,$(GENERATED_TYPES))
+
 ipv6code:
 	cp template/tree_v4.go template/tree_v6_generated.go
 	$(SED) -i -e 's/Template file./Code generated by automation. DO NOT EDIT/' template/tree_v6_generated.go
 	$(SED) -i -e 's/TreeV4/TreeV6/g' template/tree_v6_generated.go
 	$(SED) -i -e 's/treeNodeV4/treeNodeV6/g' template/tree_v6_generated.go
 	$(SED) -i -e 's/IPv4Address/IPv6Address/g' template/tree_v6_generated.go
 
-codegen: ipv6code $(addprefix codegen-,$(GENERATED_TYPES))
+generics: codegen-generics
+	# generics -> T, except in tests
+	( cd generics_tree && $(SED) -i -E -e 's/\bgenerics\b/string/g' *_test.go)
+	( cd generics_tree && $(SED) -i -E -e 's/\bgenerics\b/T/g' *.go)
+	# Each defined type should be parametrized with T
+	( cd generics_tree && $(SED) -nE 's/^type (\w+).*/\1/p' *.go \
+		| while read T; do \
+			$(SED) -i -E -e 's/\b'$$T'\b/\0[T]/g' *.go ; \
+			$(SED) -i -E -e 's/\b('$$T')\[T\]/\1[string]/g' *_test.go ; \
+		  done )
+	# Fix type definition
+	( cd generics_tree && $(SED) -i -E -e 's/^(type \w+)\[T\]/\1[T any]/' *.go)
+	# NewTreeVX function should be parametrized
+	( cd generics_tree && $(SED) -i -E -e 's/^(func NewTreeV.)/\1[T any]/' *.go)
+	( cd generics_tree && $(SED) -i -E -e 's/(NewTreeV.)\(/\1[string](/g' *_test.go)
+	# No need to cast interfaces
+	( cd generics_tree && $(SED) -i -E -e 's/\.\(string\)//g' *_test.go)
```

Comment:
@vincentbernat Whats that part of makefile responsible for?
You have used is as part of generating `generics_tree` and it still can be used somehow?

Reply:
I am unsure to understand your question. For `generics_tree`, the `codegen-%` rule is used as a first step, but further adaptations are needed. This code does that. The comments should show the different steps, specific to generics.

Reply:
Ok.. now I understood.. thanks..